src/modules/facility-loan/facility-loan.service.update-loan-expiry-date.test.ts

```diff
@@ -0,0 +1,181 @@
+    const mockSuccessfulAcbsGetLoanRequestWithResponse = (response: AcbsGetLoanByLoanIdentifierResponseDto) => {
+      when(acbsLoanServiceGetLoanByIdentifier).calledWith(portfolioIdentifier, loanIdentifier, idToken).mockResolvedValueOnce(response);
+    };
+    const mockUnsuccessfulAcbsGetLoanRequest = () => {
+      when(acbsLoanServiceGetLoanByIdentifier)
+        .calledWith(portfolioIdentifier, loanIdentifier, idToken)
+        .mockRejectedValue(new AcbsException(`Failed to update loan with identifier ${loanIdentifier} in ACBS.`));
+    };
+
+    const makeUpdateLoanExpiryDateRequestWithRequest = async (updateLoanExpiryDateRequest: UpdateLoanExpiryDateRequest) => {
+      await service.updateLoanExpiryDate(loanIdentifier, updateLoanExpiryDateRequest);
+    };
+
+    const getRequestsAndResponsesWithKnownDateParameters = ({
+      includeFinancialNextValuationDate,
+      includeCustomerUsageNextValuationDate,
+    }: {
+      includeFinancialNextValuationDate: boolean;
+      includeCustomerUsageNextValuationDate: boolean;
+    }) => {
+      const modifiedNewExpiryDate = valueGenerator.dateOnlyString();
+      const modifiedExistingDate = valueGenerator.dateTimeString();
+      const modifiedUpdateLoanExpiryDateRequest = { ...updateLoanExpiryDateRequest, expiryDate: modifiedNewExpiryDate };
+      const modifiedAcbsGetExistingLoanResponse = {
+        ...acbsGetExistingLoanResponse,
+        MaturityDate: modifiedExistingDate,
+        RateMaturityDate: modifiedExistingDate,
+        ...(includeFinancialNextValuationDate && { FinancialNextValuationDate: modifiedExistingDate }),
+        ...(includeCustomerUsageNextValuationDate && { CustomerUsageNextValuationDate: modifiedExistingDate }),
+      };
+      const modifiedAcbsUpdateLoanRequest = {
+        ...acbsUpdateLoanRequest,
+        MaturityDate: dateStringTransformations.addTimeToDateOnlyString(modifiedNewExpiryDate),
+        RateMaturityDate: dateStringTransformations.addTimeToDateOnlyString(modifiedNewExpiryDate),
+        ...(includeFinancialNextValuationDate && { FinancialNextValuationDate: dateStringTransformations.addTimeToDateOnlyString(modifiedNewExpiryDate) }),
+        ...(includeCustomerUsageNextValuationDate && {
+          CustomerUsageNextValuationDate: dateStringTransformations.addTimeToDateOnlyString(modifiedNewExpiryDate),
+        }),
+      };
+
+      return { modifiedUpdateLoanExpiryDateRequest, modifiedAcbsGetExistingLoanResponse, modifiedAcbsUpdateLoanRequest };
+    };
+
+    it('calls get loan by loan identifier endpoint with expected loan identifier', async () => {
+      mockSuccessfulAcbsGetLoanRequestWithResponse(acbsGetExistingLoanResponse);
+
+      await makeUpdateLoanExpiryDateRequestWithRequest(updateLoanExpiryDateRequest);
+
+      expect(acbsLoanServiceGetLoanByIdentifier).toHaveBeenCalledWith(portfolioIdentifier, loanIdentifier, idToken);
+      expect(acbsLoanServiceGetLoanByIdentifier).toHaveBeenCalledTimes(1);
+    });
```

Comment:
There's no harm in having this test, but you don't entirely need it because you're implicitly testing that the GET request is made with the correct parameters in other tests (otherwise your response mocking wouldn't work)

The "it's only called once" assertion isn't tested implicitly but I usually don't test that unless there's some logic going on that creates a risk of it being called more than once